client/src/components/container/portal/Attendance/index.js

```diff
@@ -24,35 +25,42 @@ class Attendance extends React.Component {
     modalDisplay: false,
   };
 
-  componentDidMount() {
-    axios
-      .get('/api/v1/event/date')
-      .then(({ data }) => {
+  async componentDidMount() {
+    try {
+      const { data } = await axios.get('/api/v1/event/date');
+      if (data.length) {
         this.setState({
           isLoaded: true,
           events: data,
         });
-      })
-      .catch((error) => {
+      } else {
         this.setState({
           isLoaded: true,
-          error,
         });
+      }
+    } catch (error) {
+      this.setState({
+        isLoaded: true,
+        error,
       });
+    }
   }
 
-  onFinish = ({ userCode }, eventCode, resetFields) => {
+  onFinish = async ({ userCode }, eventCode, resetFields) => {
     const { success, error } = this;
-    axios
-      .patch('/api/v1/attendance', {
-        userCode,
-        eventCode,
-      })
-      .then(({ data: { msg } }) => {
-        success(msg);
-        resetFields();
-      })
-      .catch(({ response: { data: { msg } } }) => error(msg));
+    try {
+      const {
+        data: { msg },
+      } = await axios.patch('/api/v1/attendance', { userCode, eventCode });
+      success(msg);
+      resetFields();
+    } catch ({
+      response: {
+        data: { msg },
+      },
+    }) {
+      error(msg);
+    }
```

Comment:
handle error without destructuring refers to Jak point as the following : 
```suggestion
   catch (err) {
      let errorMesg;

      if (err.response) {
        errorMesg = err.response.data.msg;
      } else {
        errorMesg = 'Something went wrong, please try again later';
      }
      error(errorMesg);
    }
```